spf.go

```diff
@@ -1,20 +1,52 @@
 package spf
 
 import (
-	"fmt"
+	"errors"
 	"net"
 	"strconv"
 	"strings"
+)
 
-	"github.com/miekg/dns"
+// Errors could be used for root couse analysis
+var (
+	ErrDNSTemperror      = errors.New("temporary DNS error")
+	ErrDNSPermerror      = errors.New("permanent DNS error")
+	ErrInvalidDomain     = errors.New("invalid domain name")
+	errInvalidCIDRLength = errors.New("invalid CIDR length")
+	errDNSLimitExceeded  = errors.New("limit exceeded")
 )
```

Comment:
Well, the library is not intended to be for research, and debugging is not a work you want to do on production, that is why you have such a large number of tests implemented. Which is good about the library.
Take into account that defined errors are making possible recovery logic implementation, which is quite opposite with the "human readable" errors.
All private errors used for internal logic only, while public ones intended to help using the library.
That is my view on the problem.